types/prompts/index.d.ts

```diff
@@ -92,7 +92,7 @@ declare namespace prompts {
         separator?: string;
         active?: string;
         inactive?: string;
-        choices?: Choice[];
+        choices?: Choice[] | PrevCaller<T, Choice[] | Falsy>;
```

Comment:
According to the in the docs & code in the [source repo](https://github.com/terkelg/prompts/blob/22fe104947687f9916f2dac53e658965b325870f/lib/prompts.js) `choices` can only be an array. How does this fix the types exactly?

Reply:
Choices is an array it can accept prev object the same way as other fields. For example. This prompObject fails with the current types:

```
const questions: prompts.PromptObject<string>[] = [
      {
        type: 'autocomplete',
        name: 'product',
        message: 'Select a product',
        choices: this.getChilds(''),
        suggest: (input, choices) =>
          Promise.resolve(
            choices.filter((i) =>
              i.title.toLowerCase().includes(input.toLowerCase()),
            ),
          ),
        initial: 0,
      },
      {
        type: 'autocomplete',
        name: 'application',
        message: 'Select an application',
        choices: prev => this.getChilds(prev),
        suggest: (input, choices) =>
          Promise.resolve(
            choices.filter((i) =>
              i.title.toLowerCase().includes(input.toLowerCase()),
            ),
          ),
        initial: 0,
      }
]
```
and using javascript or the modified typos works fine. In this scenario it will first ask you to select between some root options and in the second question to select an option child of this first selection, these choices are generated by `getChild()` function

Reply:
Ah found it! https://github.com/terkelg/prompts/blob/22fe104947687f9916f2dac53e658965b325870f/lib/index.js#L42

Yea makes sense. However it does this for all keys in the question object so could you update the types for all keys in `PromptObject` to use `PrevCaller`?

Reply:
test added and include prev in all the fields that I think make sense